packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -0,0 +1,470 @@
+import * as React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { get } from "lodash-es";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+import { SelectClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import { SelectHandledProps, SelectProps, SelectUnhandledProps } from "./select.props";
+import { ListboxItemProps } from "../listbox-item";
+import Listbox from "../listbox";
+import Button from "../button";
+
+export interface SelectState {
+    value: string | string[];
+    displayString: string;
+    selectedItems: ListboxItemProps[];
+    isMenuOpen: boolean;
+}
+
+class Select extends Foundation<SelectHandledProps, SelectUnhandledProps, SelectState> {
+    public static displayName: string = "Select";
+
+    public static defaultProps: Partial<SelectProps> = {
+        multiselectable: false,
+        disabled: false,
+        defaultSelection: [],
+        placeholder: "",
+    };
+
+    /**
+     * Handled props instantiation
+     */
+    protected handledProps: HandledProps<SelectHandledProps> = {
+        isMenuOpen: void 0,
+        disabled: void 0,
+        displayStringFormatterFunction: void 0,
+        form: void 0,
+        labelledBy: void 0,
+        multiselectable: void 0,
+        contentDisplayRenderFunction: void 0,
+        required: void 0,
+        managedClasses: void 0,
+        selectedItems: void 0,
+        defaultSelection: void 0,
+        onValueChange: void 0,
+        placeholder: void 0,
+        autoFocus: void 0,
+    };
+
+    private rootElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    /**
+     * constructor
+     */
+    constructor(props: SelectProps) {
+        super(props);
+
+        this.state = {
+            selectedItems: [],
+            value: "",
+            displayString: "",
+            isMenuOpen: false,
```

Comment:
I think that since `isMenuOpen` is a handled prop, we actually want state to be something where we set the prop value or default to false, `isMenuopen: this.props.isMenuOpen || false`

Reply:
Added similar logic (but with a function since we reuse this logic)